Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs

```diff
@@ -0,0 +1,118 @@
+        private readonly IFullNode fullNode;
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="loggerFactory">Factory to be used to create logger for the node.</param>
+        /// <param name="posMinting">PoS staker or null if PoS staking is not enabled.</param>
+        public MinerController(IFullNode fullNode, ILoggerFactory loggerFactory, PosMinting posMinting = null)
+        {
+            this.fullNode = fullNode;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.posMinting = posMinting;
+        }
+
+        /// <summary>
+        /// Get staking info from the miner.
+        /// </summary>
+        /// <returns>All staking info details as per the GetStakingInfoModel.</returns>
+        [Route("getstakinginfo")]
+        [HttpGet]
+        public IActionResult GetStakingInfo()
+        {
+            try
+            {
+                // checks the request is valid
+                if (!this.ModelState.IsValid)
+                {
+                    var errors = this.ModelState.Values.SelectMany(e => e.Errors.Select(m => m.ErrorMessage));
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Formatting error", string.Join(Environment.NewLine, errors));
+                }
+
+                GetStakingInfoModel model = this.posMinting != null ? this.posMinting.GetGetStakingInfoModel() : new GetStakingInfoModel();
+
+                return this.Json(model);
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Start staking.
+        /// </summary>
+        /// <param name="request">The name and password of the wallet to stake.</param>
+        /// <returns>An OKResult object that produces a status code 200 HTTP response.</returns>
+        [Route("startstaking")]
+        [HttpPost]
+        public IActionResult StartStaking([FromBody]StartStakingRequest request)
+        {
+            try
+            {
+                if (!this.ModelState.IsValid)
+                {
+                    var errors = this.ModelState.Values.SelectMany(e => e.Errors.Select(m => m.ErrorMessage));
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Formatting error", string.Join(Environment.NewLine, errors));
+                }
+
+                WalletManager walletManager = this.fullNode.NodeService<IWalletManager>() as WalletManager;
+
+                Wallet.Wallet wallet = walletManager.Wallets.FirstOrDefault(w => w.Name == request.Name);
+
+                if (wallet == null)
+                {
+                    wallet = walletManager.LoadWallet(request.Password, request.Name);
```

Comment:
I don't think you need to load the wallet, this should happen automatically on startup.
If the wallet is not found then its a wrong name so we can throw an exception.

Reply:
Ok done.